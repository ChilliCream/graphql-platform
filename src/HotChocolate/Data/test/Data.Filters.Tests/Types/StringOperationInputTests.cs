using HotChocolate.Types;
using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.Data.Filters
{
    public class StringOperationInputTests
    {
        [Fact]
        public void Create_OperationType()
        {
            // arrange
            // act
            ISchema schema = SchemaBuilder.New()
                .AddQueryType(
                    t => t
                        .Name("Query")
                        .Field("foo")
                        .Type<StringType>()
                        .Resolver("foo")
                        .Argument("test", a => a.Type<StringOperationFilterInputType>()))
                .AddFiltering()
                .Create();

            // assert
            schema.ToString().MatchSnapshot();
        }

        [Fact]
        public void Create_Implicit_Operation()
        {
            // arrange
            // act
            ISchema schema = SchemaBuilder.New()
                .AddQueryType(
                    t => t
                        .Name("Query")
                        .Field("foo")
                        .Type<StringType>()
                        .Resolver("foo")
                        .Argument("test", a => a.Type<FilterInputType<Foo>>()))
                .AddFiltering()
                .Create();

            // assert
            schema.ToString().MatchSnapshot();
        }

        [Fact]
        public void Create_Explicit_Operation()
        {
            // arrange
            // act
            ISchema schema = SchemaBuilder.New()
                .AddQueryType(
                    t => t
                        .Name("Query")
                        .Field("foo")
                        .Type<StringType>()
                        .Resolver("foo")
                        .Argument("test", a => a.Type<FooFilterInput>()))
                .TryAddConvention<IFilterConvention>(
                    (sp) => new FilterConvention(x => x.UseMock()))
                .AddFiltering()
                .Create();

            // assert
            schema.ToString().MatchSnapshot();
        }

        public class FooFilterInput : FilterInputType
        {
            protected override void Configure(IFilterInputTypeDescriptor descriptor)
            {
                descriptor.Field("string").Type<StringOperationFilterInputType>();
            }
        }

        public class Foo
        {
            public string String { get; set; } = "";

            public string? StringNullable { get; set; }
        }
    }
}
