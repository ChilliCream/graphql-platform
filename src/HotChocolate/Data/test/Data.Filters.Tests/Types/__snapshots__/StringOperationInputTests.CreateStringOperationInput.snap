schema {
  query: Query
}

type Query {
  foo(test: StringOperationInput): String
}

input StringOperationInput {
  AND: [StringOperationInput!]
  contains: String
  endsWith: String
  eq: String
  gt: String
  gte: String
  in: String
  lt: String
  lte: String
  ncontains: String
  nendsWith: String
  neq: String
  ngt: String
  ngte: String
  nin: String
  nlt: String
  nlte: String
  nstartsWith: String
  OR: [StringOperationInput!]
  startsWith: String
}

"The `String` scalar type represents textual data, represented as UTF-8 character sequences. The String type is most often used by GraphQL to represent free-form human-readable text."
scalar String
